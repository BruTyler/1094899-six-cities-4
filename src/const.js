export const PlaceType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`,
};
