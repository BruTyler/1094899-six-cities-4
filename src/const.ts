export enum PlaceType {
  APARTMENT = `Apartment`,
  ROOM = `Private room`,
  HOUSE = `House`,
  HOTEL = `Hotel`,
};

export enum SortType {
  POPULAR = `Popular`,
  PRICE_LOW = `Price: low to high`,
  PRICE_HIGH = `Price: high to low`,
  TOP_RATED = `Top rated first`,
};

export enum AuthorizationStatus {
  NO_AUTH = `NO_AUTH`,
  AUTH = `AUTH`,
};

export enum BuisnessRequirements {
  MAX_PHOTOS_PER_APARTMENT = 6,
  MAX_REVIEWS_PER_APARTMENT = 10,
  MIN_REVIEW_TEXT_LENGTH = 50,
  MAX_REVIEW_TEXT_LENGTH = 300,
};

export enum AppRoute {
  AUTH = `/login`,
  ROOT = `/`,
  FAVORITES = `/favorites`,
  PROPERTY_WITH_ID = `/offer/:id`,
};

export enum ApartmentEnvironment {
  FAVORITE = `favorites`,
  NEARBY_PLACES = `near-places`,
  MAIN_WINDOW = `cities`
};

export enum MapEnvironment {
  NEARBY_PLACES = `property__map`,
  MAIN_WINDOW = `cities__map`
};
