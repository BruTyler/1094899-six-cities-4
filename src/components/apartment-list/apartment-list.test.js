import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import ApartmentList from './apartment-list.jsx';
import NameSpace from '../../reducer/name-space.js';
import {AuthorizationStatus, ApartmentEnvironment} from '../../const.js';

const APARTMENTS = [
  {
    id: 0,
    type: `Apartment`,
    description: `description0`,
    rating: 0.1,
    price: 1,
    isPremium: true,
    isFavourite: true,
    photo: `img0.jpg`
  },
  {
    id: 1,
    type: `Private room`,
    description: `description1`,
    rating: 1,
    price: 2,
    isPremium: false,
    isFavourite: false,
    photo: `img1.jpg`
  }
];
const mockStore = configureStore();

describe(`<ApartmentList /> render suit`, () => {
  it(`<ApartmentList /> render apartment list`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
    });
    const generatedTree = renderer.create(
        <Provider store={store}>
          <ApartmentList
            parentBox={ApartmentEnvironment.MAIN_WINDOW}
            apartmentList={APARTMENTS}
          />
        </Provider>
    ).toJSON();

    expect(generatedTree).toMatchSnapshot();
  });
});
