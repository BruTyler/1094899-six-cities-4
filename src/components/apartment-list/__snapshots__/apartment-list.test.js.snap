// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApartmentList /> render suit <ApartmentList /> render apartment list 1`] = `
<div
  className="cities__places-list cities__list places__list tabs__content"
>
  <article
    className="cities__place-card cities__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="description0"
          className="place-card__image"
          height="200"
          src="img0.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            1
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In
             bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "0%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          description0
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card cities__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="description1"
          className="place-card__image"
          height="200"
          src="img1.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            2
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button  place-card__bookmark-button--active"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To
             bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "20%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          description1
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Private room
      </p>
    </div>
  </article>
</div>
`;
